From ab5c20be4a987d1cd6d2e472634f5e1b9c11211f Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 2 Dec 2021 10:05:12 +1100
Subject: [PATCH 2/5] do not build vendored abseil & libprotobuf-lite

ensure we can use shared builds of libprotobuf also on windows
---
 src/CMakeLists.txt         | 5 +++++
 third_party/CMakeLists.txt | 5 +----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fbdf238..2b8aefa 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -71,6 +71,11 @@ if (SPM_PROTOBUF_PROVIDER STREQUAL "internal")
   include_directories(builtin_pb)
 elseif (SPM_PROTOBUF_PROVIDER STREQUAL "package")
   find_package(Protobuf REQUIRED)
+  if (MSVC)
+   add_definitions("/DPROTOBUF_USE_DLLS")
+  else()
+   add_definitions("-DPROTOBUF_USE_DLLS")
+  endif()
   include_directories(${Protobuf_INCLUDE_DIRS})
   protobuf_generate_cpp(SPM_PROTO_SRCS SPM_PROTO_HDRS sentencepiece.proto)
   protobuf_generate_cpp(SPM_MODEL_PROTO_SRCS SPM_MODEL_PROTO_HDRS sentencepiece_model.proto)
diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index d00ecba..3096702 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -1,4 +1 @@
-include_directories(absl/strings darts_clone esaxx protobuf-lite)
-
-
-
+include_directories(darts_clone esaxx)
